prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <limits.h>
+
+int generate_primes(int num) {
+    int cnt = 0;
+    printf("Prime Factors: ");
```

Comment:
Please reply in proper thread.

> n has to be the upper bound, num has to be the number

That is what I am trying to tell you from the beginning. The number can itself be an upper bound for the number of possible primes for that number.